avm/res/app/session-pool/main.bicep

```
@@ -0,0 +1,272 @@
+metadata name = 'Container App Session Pool'
+metadata description = 'This module deploys a Container App Session Pool.'
+
+@description('Required. Name of the Container App Session Pool.')
+param name string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Required. The container type of the sessions.')
+@allowed(['PythonLTS', 'CustomContainer'])
+param containerType string
+
+@description('Optional. Custom container definitions. Only required if containerType is CustomContainer.')
+param containers sessionContainerType
```

As a consequence of the above suggestion
```suggestion
param containers sessionContainerType[]?
```